streams/lms_metadata.py

```diff
@@ -16,29 +16,34 @@ class LMSMetadataReader:
   # meta_ref is probably an unneccessary variable to pass as an arg since it's obscured from the user, but we can eventually make it an optional setting for the user
   def __init__(self, name: str, meta_ref: Optional[int] = 2, dump: Optional[bool] = False):
     self.player_name = name
-    self.IP = None
+    self.locale = None # locale replaces IP, it is a concatenation of both the IP and the port in {IP}:{port} format
     self.meta_ref_rate = meta_ref
     self.connected = False
     self.dump = dump
 
   def connect(self):
     """Discovers LMS Player and then requests metadata repetitively"""
-    f = open(f"lms_{str(self.player_name).replace(' ', '_')}_metadata.json", 'wt', encoding='utf-8')
-    json.dump({'track': 'Loading...', 'artist': 'Loading...', 'album': 'Loading...', 'image_url': 'static/imgs/lms.png'}, f, indent = 2)
-    f.close()
+    with open(f"lms_{str(self.player_name).replace(' ', '_')}_metadata.json", 'wt', encoding='utf-8') as f:
+      json.dump({'track': 'Loading...', 'artist': 'Loading...', 'album': 'Loading...', 'image_url': 'static/imgs/lms.png'}, f, indent = 2)
     x = 0
 
-    # Much faster method of connecting to the metadata server using code from: https://github.com/ralph-irving/squeezelite/blob/master/tools/find_server.c
-    IPFind = subprocess.run('streams/find_server', capture_output=True, text=True)
-    # Uses re.search because find_server.c spits out as '{Hostname}:{port} ({IP})', so I scrape the data from inbetween the parentheses to get the IP
-    self.IP = re.search(r"\((.*?)\)", IPFind.stdout).group(1)
-
 
     # When not connected, search for player to connect to by the proper name
     while not self.connected:
       try:
+        # Much faster method of connecting to the metadata server using code from: https://github.com/ralph-irving/squeezelite/blob/master/tools/find_server.c
+        ip_find = subprocess.run(['streams/find_server'], capture_output=True, text=True)
+        print(f'STDOUT: {ip_find.stdout}')
+        # Uses re.search because find_server.c spits out as '{Hostname}:{port} ({IP})', so I scrape the data from inbetween the parentheses to get the IP
+        ip = re.search(r'\((.*?)\)', ip_find.stdout).group(1)
+        print(f'IP: {ip}')
+        port = re.search(r':(\d{4})', ip_find.stdout).group(1)
+        print(f'PORT: {port}')
+        self.locale = f"{ip}:{port}"
+
```

Comment:
i would hide the stdout printout under a debug flag specified by the command line like
```
if self.debug:
   print(f'find_server output: {ip_find.stdout}')
```
condense the rest of the printouts to something like print('Found LMS server: {ip}:{port}')